### PR TITLE
[Common Lisp] Add reference document for cons type.

### PR DESCRIPTION
**Pull request by verdammelt**

### CODE REVIEW

File: languages/common-lisp/reference/types/cons.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Cons Cells
+
+A [cons cell][cons] (also known as a dotted pair due its printed form) is a data structure consisting of a pair of two objects. Cons cells are created with the function `cons` and the items of a cons cell are accessed with `car` and `cdr`.
```
Comment (TheLostLambda) [inline languages/common-lisp/reference/types/cons.md:3]: Looks excellent! We could mention `first` and `rest`? Not too fussed either way!

File: languages/common-lisp/reference/types/cons.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+(The names of the accessor methods are a [historical artifact][why-car-cdr]. The cons cells of the original Lisp implementation were implemented in registers of the IBM 704 computer and the operations used to access them were the `CAR` operation to access the "*C*ontents of the *A*dress part of the *Register" and `CDR` operation to access the "*C*ontents of the *D*ecrement part of the *R\*egister".)
```
Comment (TheLostLambda) [inline languages/common-lisp/reference/types/cons.md:5]: Part of this got a little muddled by markdown (there was a backslash hiding in there). I also think that bold is a bit clearer than italics here. I also added a missing "d" to address

I'd suggest:
```
"**C**ontents of the **A**ddress part of the **R**egister" and `CDR` operation to access the "**C**ontents of the **D**ecrement part of the **R**egister".
```
Which looks like:

"**C**ontents of the **A**ddress part of the **R**egister" and `CDR` operation to access the "**C**ontents of the **D**ecrement part of the **R**egister".

File: languages/common-lisp/reference/types/cons.md (hunk 1 above)
Comment (verdammelt) [reply languages/common-lisp/reference/types/cons.md:3]: I vassilated on this one. `first` and `rest` to me feel like things that are specifically for lists.

But i will add them parenthetically.

File: languages/common-lisp/reference/types/cons.md (hunk 2 above)
Comment (verdammelt) [reply languages/common-lisp/reference/types/cons.md:5]: For more than a decade before Markdown was 'invented' there was a text markup common in email & usenet. It used * for emphasis, / for italics and _ for underline. My brain and fingers have not yet come to terms with the 'new' standard.

File: languages/common-lisp/reference/types/cons.md (hunk 1 above)
Comment (TheLostLambda) [reply languages/common-lisp/reference/types/cons.md:3]: Hmm, I somewhat agree actually. They probably don't need to be there. Up to you!

File: languages/common-lisp/reference/types/cons.md (hunk 2 above)
Comment (TheLostLambda) [reply languages/common-lisp/reference/types/cons.md:5]: Haha, I write a bunch in org-mode and it's the same. It makes markdown feel a bit unintuitive!

**Pull request merged**